test/cloud-canary/mzcompose.py

```diff
@@ -0,0 +1,112 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+import os
+import ssl
+import time
+import urllib.parse
+
+import pg8000
+
+from materialize.mzcompose import Composition, _wait_for_pg
+from materialize.mzcompose.services import Materialized, Mz, Testdrive
+from materialize.ui import UIError
+
+REGION = "aws/us-east-1"
+ENVIRONMENT = "staging"
+USERNAME = "infra+nightly-canary@materialize.com"
+APP_PASSWORD = os.environ["NIGHTLY_CANARY_APP_PASSWORD"]
+VERSION = "devel-" + os.environ["BUILDKITE_COMMIT"]
+
+SERVICES = [
+    Materialized(),
+    Testdrive(),
+    Mz(
+        region=REGION,
+        environment=ENVIRONMENT,
+        username=USERNAME,
+        app_password=APP_PASSWORD,
+    ),
+]
+
+
+def workflow_default(c: Composition) -> None:
+    """Deploy the current source to the cloud and run a smoke test."""
+
+    print("Obtaining mz_version() string from local instance ...")
+    c.start_and_wait_for_tcp(services=["materialized"])
+    c.wait_for_materialized()
+    local_version = c.sql_query("SELECT mz_version();")[0][0]
+
+    print(f"Shutting down region {REGION} ...")
+    c.run("mz", "region", "disable", REGION)
+
+    test_failed = True
+    try:
+        print(f"Enabling region using Mz version {VERSION} ...")
+        try:
+            c.run("mz", "region", "enable", REGION, "--version", VERSION)
+        except UIError:
+            # Work around https://github.com/MaterializeInc/materialize/issues/17219
+            pass
+
+        time.sleep(10)
+
+        print("Obtaining hostname of cloud instance ...")
+        region_status = c.run("mz", "region", "status", REGION, capture=True)
+        sql_line = region_status.stdout.split("\n")[1]
+        cloud_hostname = sql_line.split(":")[1].strip()
+        assert "materialize.cloud" in cloud_hostname
+
+        materialize_url = f"postgres://{urllib.parse.quote(USERNAME)}:{urllib.parse.quote(APP_PASSWORD)}@{cloud_hostname}:6875"
```

Comment:
I backed out the testdrive change because `tokio_postgres` works fine if you URL-escape the username and password!

Reply:
This will mean the URL contain sensitive information, so it should never appear in the logs. This is a tall requirement to track going forward. I would say at least the password as an env variable would have been safer.

I wish Buildkite had the opportunity to designate a particular step in a pipeline as non-private, but I can not find such a functionality. As a compromise, should we consider a non-private dedicated nightly-scheduled pipeline for any cloud-interfacing jobs -- I assume this job will be the first of many?

Reply:
> As a compromise, should we consider a non-private dedicated nightly-scheduled pipeline for any cloud-interfacing jobs -- I assume this job will be the first of many?

Oh, I suppose.